celery/worker/control.py

```diff
@@ -146,6 +147,49 @@ def revoke(state, task_id, terminate=False, signal=None, **kwargs):
     #     Outside of this scope that is a function.
     # supports list argument since 3.1
     task_ids, task_id = set(maybe_list(task_id) or []), None
+    return _revoke(state, task_ids, terminate, signal, **kwargs)
+
+
+@control_command(
+    variadic='header',
+    signature='[header1 [header2 [... [headerN]]]]',
+)
+def revoke_by_stamped_header(state, header, terminate=False, signal=None, **kwargs):
+    """Revoke task by header (or list of headers).
+
+    Keyword Arguments:
+        terminate (bool): Also terminate the process if the task is active.
+        signal (str): Name of signal to use for terminate (e.g., ``KILL``).
+    """
+    # pylint: disable=redefined-outer-name
+    # XXX Note that this redefines `terminate`:
+    #     Outside of this scope that is a function.
+    # supports list argument since 3.1
+    headers, header = maybe_list(header) or [header], None
```

Comment:
The header argument can also be a list of `header=value` when received from the CLI.
We need to parse the value and check if it's either a dictionary or a list of strings.